src/rules/valid-properties.ts

```diff
@@ -0,0 +1,59 @@
+import {
+	validateAuthor,
+	validateBundleDependencies,
+	validateConfig,
+	validateLicense,
+	validateScripts,
+	validateType,
+} from "package-json-validator";
+
+import type { PackageJsonRuleModule } from "../createRule.js";
+
+import {
+	createSimpleValidPropertyRule,
+	ValidationFunction,
+} from "../utils/createSimpleValidPropertyRule.js";
+
+interface ValidPropertyOptions {
+	aliases: string[];
+	validator: ValidationFunction;
+}
+
+// List of all properties we want to create valid- rules for,
+// in the format [propertyName, validationFunction | validPropertyOptions]
+const properties = [
+	["author", validateAuthor],
+	[
+		"bundleDependencies",
+		{
+			aliases: ["bundledDependencies"],
+			validator: validateBundleDependencies,
+		},
+	],
+	["config", validateConfig],
+	["license", validateLicense],
+	["scripts", validateScripts],
+	["type", validateType],
+	// TODO: More to come!
+] satisfies [string, ValidationFunction | ValidPropertyOptions][];
+
+/** All basic valid- flavor rules */
+export const rules = properties.reduce<Record<string, PackageJsonRuleModule>>(
+	(acc, [propertyName, validationFunctionOrOptions]) => {
+		let validationFunction: ValidationFunction;
+		let aliases: string[] = [];
+		if (typeof validationFunctionOrOptions === "object") {
+			validationFunction = validationFunctionOrOptions.validator;
+			aliases = validationFunctionOrOptions.aliases;
+		} else {
+			validationFunction = validationFunctionOrOptions;
+		}
+		acc[`valid-${propertyName}`] = createSimpleValidPropertyRule(
+			propertyName,
+			validationFunction,
+			aliases,
+		);
+		return acc;
+	},
+	{},
+);
```

Comment:
[Refactor] I personally find the combo of `const [ ... ] = ... ? ... : ...` pattern and the `Object.fromEntries` a little easier to read. Absolutely not blocking though, just a mild suggestion.

```suggestion
export const rules = Object.fromEntries(
	properties.map(([propertyName, validationFunctionOrOptions]) => {
		const [aliases, validationFunction] =
			typeof validationFunctionOrOptions === "object"
				? [
						validationFunctionOrOptions.aliases,
						validationFunctionOrOptions.validator,
					]
				: [[], validationFunctionOrOptions];

		return [
			`valid-${propertyName}`,
			createSimpleValidPropertyRule(
				propertyName,
				validationFunction,
				aliases,
			),
		];
	}),
);
```

Reply:
I think memory and speed characteristics are (minuscule-y) better for the reduce approach.  Unless you feel strongly about it, I'm inclined to leave the reduce versions of these?